internal/executor/rpc/rpc.go

```diff
@@ -298,3 +298,38 @@ func (r *RPC) ReportAgentSignal(ctx context.Context, req *api.ReportAgentSignalR
 
 	return &empty.Empty{}, nil
 }
+
+func (r *RPC) ReportAnnotations(ctx context.Context, req *api.ReportAnnotationsCommandRequest) (*empty.Empty, error) {
+	_, err := r.build.GetTaskFromIdentification(req.TaskIdentification, r.clientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, annotation := range req.Annotations {
+		if annotation.FileLocation == nil {
+			continue
+		}
+
+		var mappedLevel echelon.AnnotationLevel
+
+		switch annotation.Level {
+		case api.Annotation_NOTICE:
+			mappedLevel = echelon.AnnotationLevelNotice
+		case api.Annotation_WARNING:
+			mappedLevel = echelon.AnnotationLevelWarning
+		case api.Annotation_FAILURE:
+			mappedLevel = echelon.AnnotationLevelError
+		}
+
+		r.logger.Annotation(&echelon.Annotation{
```

Comment:
I don't think echelon should know anything about the annotations. From it's perspective it's just a text log entry.

Maybe add the annotation notion to the `FoldableLogsRenderer` and check here if `r.logger` is an instance of it and then print the annotation?

Reply:
Implemented direct annotation rendering in https://github.com/cirruslabs/cirrus-cli/pull/500/commits/489c9571694a4c6f7655e7d662dd7148b2a75604.